test/controllers/ReferenceDataControllerSpec.scala

```diff
@@ -50,27 +54,94 @@ class ReferenceDataControllerSpec extends SpecBase with GuiceOneAppPerSuite with
     FakeRequest(POST, routes.ReferenceDataController.post().url)
-      .withJsonBody(Json.obj())
+      .withRawBody(ByteString.apply(byteArray))
+
+  def compress(input: Array[Byte]): Array[Byte] = {
+    val bos  = new ByteArrayOutputStream(input.length)
+    val gzip = new GZIPOutputStream(bos)
+    gzip.write(input)
+    gzip.close()
+    val compressed = bos.toByteArray
+    bos.close()
+    compressed
+  }
 
   "post" - {
-    "returns Ok when the data has been processed" in {
+    "returns Ok when the data has been processed with a valid GZipped Json body" in {
       when(mockReferenceDataService.insert(any())).thenReturn(Future.successful(DataProcessingSuccessful))
 
-      val result = route(app, fakeRequest).value
+      val json =
+        """
+          |{
+          |   "messageInformation": {
+          |     "messageID": "74bd0784-8dc9-4eba-a435-9914ace26995",
+          |     "snapshotDate": "2020-07-06"
+          | }
+          |}
+          |""".stripMargin
+
+      val compressedJson = compress(json.getBytes)
+
+      val result = route(app, fakeRequest(compressedJson)).value
 
       status(result) mustBe Status.ACCEPTED
     }
 
     "returns with an Internal Server Error when the data was not processed successfully" in {
       when(mockReferenceDataService.insert(any())).thenReturn(Future.successful(DataProcessingFailed))
 
-      val result = route(app, fakeRequest).value
+      val json =
+        """
+          |{
+          |   "messageInformation": {
+          |     "messageID": "74bd0784-8dc9-4eba-a435-9914ace26995",
+          |     "snapshotDate": "2020-07-06"
+          | }
+          |}
+          |""".stripMargin
+
+      val compressedJson = compress(json.getBytes)
+
+      val result = route(app, fakeRequest(compressedJson)).value
 
       status(result) mustBe Status.INTERNAL_SERVER_ERROR
       contentAsJson(result) mustBe Json.toJsObject(ResponseErrorMessage(OtherError, None))
     }
 
+    "returns Internal Server Error if request body is not in compressed GZip format" in {
+
+      val json =
+        """
+          |{
+          |   "messageInformation": {
+          |     "messageID": "74bd0784-8dc9-4eba-a435-9914ace26995",
+          |     "snapshotDate": "2020-07-06"
+          | }
+          |}
+          |""".stripMargin
+
+      val result = route(app, fakeRequest(json.getBytes)).value
+
+      status(result) mustBe Status.INTERNAL_SERVER_ERROR
+    }
+
+    "returns Internal Server Error if request body is compressed however Json is invalid" in {
+
+      val invalidJson = "Invalid Json"
+
+      val result = route(app, fakeRequest(invalidJson.getBytes)).value
+
+      status(result) mustBe Status.INTERNAL_SERVER_ERROR
+    }
+
+//    "returns Internal Server Error if request body is larger than memory threshold" in {
+//
+//      val result = route(app, fakeRequest(???)).value
+//
+//      status(result) mustBe Status.INTERNAL_SERVER_ERROR
+//    }
```

Comment:
I don't think we should aim to test this in the unit tests, this tuning can be done with performance testing